### PR TITLE
Feature/monitor register

### PR DESCRIPTION
**Pull request by wjeam**
Added monitor registration, made some changed to the UX

### CODE REVIEW

**Comment by wjeam** [comment]
In TopBar.js, I hardcoded the options for the dropdown menu at the top right. We will need to modify it eventually to make it easier to add more options. As of now, it is not a priority as we don't have many options.

File: backend/src/main/java/com/team4/backend/mapping/MonitorMapper.java [hunk 1]
```diff
@@ -1,20 +1,33 @@
...
 
+import com.team4.backend.dto.MonitorDto;
 import com.team4.backend.model.Monitor;
 
 public abstract class MonitorMapper {
 
-    //TODO add tests
+    public static Monitor toEntity(MonitorDto monitorDto) {
+        return Monitor.monitorBuilder()
+                .id(monitorDto.getId())
```
Comment (0xD34DC0DE) [inline backend/src/main/java/com/team4/backend/mapping/MonitorMapper.java:10]: id coming from frontend shouldn't be trusted -> remove id mapping inside toEntity

File: backend/src/test/java/com/team4/backend/controller/MonitorControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        monitorDto.setId(null); // Frontend gives null id
+
+        Monitor monitor = MonitorMockData.getMockMonitor();
+
+        when(monitorService.registerMonitor(any(Monitor.class)))
+                .thenReturn(
+                        Mono.just(monitor)
+                                .map(s -> {
+                                    s.setId("615259e03835be1f53bd49e4");
+                                    return s;
```
Comment (0xD34DC0DE) [inline backend/src/test/java/com/team4/backend/controller/MonitorControllerTest.java:53]: s stands for ??? (might want a m for monitor instead)

File: backend/src/test/java/com/team4/backend/mapping/MonitorMapperTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    @Test
+    void mapDtoToEntity() {
+        //ARRANGE
+        MonitorDto dto = MonitorMockData.getMockMonitorDto();
+
+        //ACT
+        Monitor entity = MonitorMapper.toEntity(dto);
+
+        //ASSERT
+        assertEquals(dto.getId(), entity.getId());
```
Comment (0xD34DC0DE) [inline backend/src/test/java/com/team4/backend/mapping/MonitorMapperTest.java:42]: if you fix the dto -> entity id, you will need to change the assert here too

File: frontend/src/components/Content.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+          <>
+            Hello {userInfo.role.toLowerCase()}, {userInfo.email}{" "}
+          </>
+        ) : (
+          "Hello visitor."
+        )}
+      </Typography>
+    </Container>
+  );
+};
```
Comment (0xD34DC0DE) [inline frontend/src/components/Content.js:21]: Where are the child components going to be displayed ?
The content component should be a container (the content area) to the currently displayed work space.
See UserInfoStore for child component example.
Call me on discord for questions.

**Review by 0xD34DC0DE — CHANGES_REQUESTED** [review]

**Review by 0xD34DC0DE — APPROVED** [review]

**Pull request merged**